R/readHouseholdSurveys.R

```diff
@@ -12,24 +12,48 @@
 #' @importFrom quitte as.quitte
 #' @importFrom magclass as.magpie
 
-readHouseholdSurveys <- function(subtype = c("cooking", "lighting")) {
+readHouseholdSurveys <- function(subtype = c("cooking", "lighting", "appliances")) {
 
   # read data sheet
   data <- switch(subtype,
-                 cooking  = read.xlsx("Cooking_fuels.xlsx"),
-                 lighting = read.xlsx("Lighting_fuels.xlsx"))
+                 cooking    = read.xlsx("Cooking_fuels.xlsx"),
+                 lighting   = read.xlsx("Lighting_fuels.xlsx"),
+                 appliances = read.xlsx("Appliances.xlsx"))
+
+  # region mapping
+  regionmap <- toolGetMapping("regionmappingHouseholdSurveys.csv",
+                              type = "regional",
+                              where = "mredgebuildings")
 
   # standardize column names
   colnames(data) <- tolower(colnames(data))
 
   # change country codes
   data <- data %>%
-    rename("period" = "year",
-           "region" = "country_long") %>%
-    pivot_longer(cols = -all_of(c("region", "period")), names_to = "carrier", values_to = "value") %>%
-    mutate(carrier = sub("\\.", "_", .data$carrier),
-           enduse = subtype,
-           period = ifelse(is.na(.data$period), median(.data$period, na.rm = TRUE), .data$period)) %>%
+    mutate(region = toolCountry2isocode(country = .data$country_long,
+                                        mapping = setNames(as.list(regionmap$regionTarget),
+                                                           regionmap$region))) %>%
+    select(-"country_long") %>%
+    rename("period" = "year")
+
+  data <- if (subtype == "appliances") {
+    data %>%
+      pivot_longer(cols = -all_of(c("region", "period")),
+                   names_to = "variable",
+                   values_to = "value") %>%
+      mutate(variable = sub(".01", "", .data$variable),
+             value = replace_na(.data$value, 0))
+  } else {
+    data %>%
+      pivot_longer(cols = -all_of(c("region", "period")),
+                   names_to = "carrier",
+                   values_to = "value") %>%
+      mutate(carrier = sub(".\\", "", .data$carrier),
+             enduse  = subtype)
+  }
```

Comment:
I though this could be generalised much more. Then I realised you create the end use column only in one case. This seems a bit inconsistent, right? No need to bother about this comment if it requires too much effort.